pkg/network/setup/netpod/discoverbridge.go

```diff
@@ -121,14 +121,21 @@ func translateNmstateToNetlinkRoutes(otherRoutes []nmstate.Route) ([]vishnetlink
 // filterRoutesByNonLocalDestination filters out local routes (the destination is of the local link).
 // Default routes should not be filter out.
 func filterRoutesByNonLocalDestination(linkRoutes []nmstate.Route, addr *vishnetlink.Addr) ([]nmstate.Route, error) {
+	_, localNetwork, err := net.ParseCIDR(addr.String())
+	if err != nil {
+		return nil, err
+	}
+
+	localNetworkStr := localNetwork.String()
+
 	var otherRoutes []nmstate.Route
 	for _, route := range linkRoutes {
 		_, dstIPNet, perr := net.ParseCIDR(route.Destination)
 		if perr != nil {
 			return nil, perr
 		}
 		isDefaultRoute := route.Destination == nmstate.DefaultDestinationRoute(vishnetlink.FAMILY_V4).String()
-		localDestination := !isDefaultRoute && dstIPNet.Contains(addr.IP)
+		localDestination := !isDefaultRoute && dstIPNet.String() == localNetworkStr
```

Comment:
what about narrower subnet ? or it can't happen

Reply:
> What about narrower subnet ? it won't be filtered, it is ok right ? Maybe worth to add an unit test for that, and to mention it on PR desc ?

The commit message and PR description state: "Filter only non-default routes whose destination **exactly matches** the local subnet of the pod's NIC."
This means that narrower subnets will not be filtered and will be passed to the VM.

Since at the moment we are not aware of a use case that requires passing the narrower subnet, I don't think its worth adding such a test case.

Reply:
I saw that, just the fact "wider" was emphasized and then it goes "general" caused me to ask that, anyhow we are good, thanks